repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -0,0 +1,159 @@
+import json
+from collections import namedtuple
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import InstalledRedHatSignedRPM, RpmTransactionTasks, RepositoriesSetupTasks
+
+
+REPOSITORIES_BLACKLIST = ('rhel8-buildroot', 'rhel8-crb')
+# FIXME: this mapping is not complete and will need to be manually updated frequently
+REPOSITORIES_MAPPING = {
+    'rhel8-appstream': 'rhel-8-for-x86_64-appstream-htb-rpms',
+    'rhel8-baseos': 'rhel-8-for-x86_64-baseos-htb-rpms'}
+
+
+Event = namedtuple('Event', ['action', 'in_pkgs', 'out_pkgs'])
+
+
+def parse_action(action_id):
+    """ Get action label from PES event's action id """
+    labels = ('Present', 'Removed', 'Deprecated', 'Replaced', 'Split', 'Merged', 'Moved', 'Renamed')
+
+    if action_id < 0 or action_id >= len(labels):
+        raise ValueError
+
+    return labels[action_id]
+
+
+def parse_packageset(packageset):
+    """ Get packages and repositories data from PES event's package set """
+    if not packageset:
```

Comment:
well while I get the check, according to your usage of parse_packageset you can remove this check. You do control the input, and you do not pass None but only empty dicts. Which this code anyway will result with an empty dict too.